ts/src/config.ts

```diff
@@ -87,8 +87,8 @@ export interface ProfilerConfig extends AuthenticationConfig {
   projectId?: string;
   logLevel: number;
   serviceContext: {service: string; version?: string;};
-  instance: string;
-  zone: string;
+  instance?: string;
```

Comment:
I would keep the type and skip sending if the string is empty - what's wrong with that? We don't want to send empty strings either and I am not sure the current code does not?

Reply:
Right now, the code will skip sending these if empty.
I've added a test for that now.